contracts/hydro/src/testing_cw721.rs

```diff
@@ -0,0 +1,2165 @@
+        minimum_atom_liquidity_request: Uint128::zero(),
+    };
+    let proposal_res = execute(deps.as_mut(), env.clone(), info.clone(), proposal_msg);
+    assert!(
+        proposal_res.is_ok(),
+        "Failed to create proposal: {:?}",
+        proposal_res
+    );
+
+    // Vote on proposal
+    let vote_msg = ExecuteMsg::Vote {
+        tranche_id: 1,
+        proposals_votes: vec![ProposalToLockups {
+            proposal_id: 0,
+            lock_ids: vec![0],
+        }],
+    };
+    let vote_res = execute(deps.as_mut(), env.clone(), info.clone(), vote_msg);
+    assert!(vote_res.is_ok(), "Failed to vote: {:?}", vote_res);
+
+    // verify user's vote worked
+    let vote_query_res = query_all_votes(deps.as_ref(), 0, 100);
+    assert!(
+        vote_query_res.is_ok(),
+        "Vote query should not fail: {:?}",
+        vote_query_res
+    );
+    let votes = vote_query_res.unwrap().votes;
+    assert_eq!(1, votes.len());
+    let vote = votes.first().unwrap();
+    assert_eq!(vote.sender_addr, info.sender);
+    assert_eq!(0, vote.lock_id);
+    assert_eq!(0, vote.vote.prop_id);
+
+    let recipient = get_address_as_str(&deps.api, "recipient");
+    let token_id = "0".to_string(); // First lock ID is 0
+    let transfer_msg = ExecuteMsg::TransferNft {
+        recipient: recipient.clone(),
+        token_id: token_id.clone(),
+    };
+
+    // Execute the message using the contract's execute function
+    let transfer_res = execute(deps.as_mut(), env.clone(), info.clone(), transfer_msg);
+
+    // Verify the response
+    assert!(
+        transfer_res.is_ok(),
+        "Failed to transfer NFT: {:?}",
+        transfer_res
+    );
+    let transfer_res = transfer_res.unwrap();
+
+    // Check that action is set correctly
+    assert!(transfer_res
+        .attributes
+        .iter()
+        .any(|attr| attr.key == "action" && attr.value == "transfer_nft"));
+
+    // Check that the owner does not have the lock anymore
+    let res = query_specific_user_lockups(
+        &deps.as_ref(),
+        &env,
+        info.sender.to_string(),
+        vec![0], // Query only the first lockup
+    );
+
+    assert!(res.is_ok());
+    let res = res.unwrap();
+    assert_eq!(0, res.lockups.len());
+
+    // Check that the new owner (recipient) has the lock
+    let res = query_specific_user_lockups(
+        &deps.as_ref(),
+        &env,
+        recipient.to_string(),
+        vec![0], // Query only the first lockup
+    );
+    assert!(res.is_ok());
+    let res = res.unwrap();
+    assert_eq!(1, res.lockups.len());
+    assert_eq!(0, res.lockups[0].lock_entry.lock_id);
+    assert_eq!(1000, res.lockups[0].lock_entry.funds.amount.u128());
+    assert_eq!(ST_ATOM_ON_NEUTRON, res.lockups[0].lock_entry.funds.denom);
+
+    // Also check via the Owner Of query
+    let res = query_owner_of(deps.as_ref(), env, token_id, None);
+    assert!(res.is_ok());
+
+    let res = res.unwrap();
+    assert_eq!(res.owner, recipient.to_string());
```

Comment:
should there be a check here to verify the vote?

```suggestion
    // verify the transferred lockup has still voted
    let vote_query_res = query_all_votes(deps.as_ref(), 0, 100);
    assert!(
        vote_query_res.is_ok(),
        "Vote query should not fail: {:?}",
        vote_query_res
    );
    let votes = vote_query_res.unwrap().votes;
    assert_eq!(1, votes.len());
    let vote = votes.first().unwrap();
    assert_eq!(vote.sender_addr.to_string(), recipient);
    assert_eq!(0, vote.lock_id);
    assert_eq!(0, vote.vote.prop_id);
```